main.tf

```diff
@@ -73,22 +97,43 @@ resource "aws_cloudwatch_log_group" "default" {
 
 module "container_definition" {
   source  = "cloudposse/ecs-container-definition/aws"
-  version = "0.58.1"
+  version = "0.58.3"
 
   container_name               = module.ecs_label.id
-  container_image              = "${var.app_image_repository}:${local.image_tag}"
+  container_image              = "${module.ecr.repository_url}:${local.image_tag}"
   container_memory             = var.container_memory
   container_memory_reservation = var.container_memory_reservation
   container_cpu                = var.container_cpu
   start_timeout                = var.container_start_timeout
   stop_timeout                 = var.container_stop_timeout
   healthcheck                  = local.healthcheck
-  map_environment              = var.container_map_environment
-  map_secrets                  = var.container_map_secrets
-  port_mappings                = local.port_mappings
-  ulimits                      = var.ulimits
-  working_directory            = var.working_directory
-  docker_labels                = local.docker_labels
+  map_environment = merge(var.container_map_environment, {
```

Comment:
same as above, we should rather append var.container_map_environment instead of prepending it. That way we can overwrite values easily instead of having our passed values overwritten by a fixed local map.